src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,142 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node<T> head = null;
+    private Node<T> tail = null;
+
+    private int size = 0;
```

Comment:
int type = 0 by default
```suggestion
    private int size;
```